pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -214,6 +214,13 @@ func (admitter *VMsAdmitter) applyInstancetypeToVm(vm *v1.VirtualMachine) []meta
 func (admitter *VMsAdmitter) authorizeVirtualMachineSpec(ar *admissionv1.AdmissionRequest, vm *v1.VirtualMachine) ([]metav1.StatusCause, error) {
 	var causes []metav1.StatusCause
 
+	// Skip DataVolumeTemplates validation if we want to delete the vm
+	// This allows the vm controller to remove the finalizers
+	// also when, for example, the datasource is already gone.
+	if vm.DeletionTimestamp != nil && len(vm.Finalizers) == 0 {
+		return causes, nil
+	}
+
```

Comment:
nit: I know that data volumes is the only subject being tested in this function, but more validations can be added in the future. WDYT about something like:

```go
if vm.DeletionTimestamp == nil || len(vm.Finalizers) > 0 {
   for idx, dataVolume := range vm.Spec.DataVolumeTemplates {
      // ...
   }
}
```

but maybe I'm over reacting here :)

Reply:
Thanks for the suggestion! Done! PTAL :)